packages/dropdown/src/react/__stories__/index.story.js

```diff
@@ -578,3 +580,167 @@ storiesOf('portal', module).add('position', _ => (
     ultrices eros in cursus turpis massa.
   </div>
 ))
+const DropdownWithIcon = forwardRef(
```

Comment:
So this component maps to the refactor with the addition of an icon prop. After finishing everything I just realized I could just add the prop to the existing index.js in the react dir in Dropdown and it wouldn't be horrible prop creep.

Reply:
love the composability. any thoughts on ways we could avoid destructing the whole context and then passing the props on. maybe groups of like `menuProps`, `buttonProps`, etc. 

ie https://github.com/downshift-js/downshift/tree/master/src/hooks/useSelect#returned-props

Reply:
The downshift pattern is interesting.  I could go along with it, but I don't know that we need that extra abstraction.

I would go the opposite route to cut down on destructuring:

```
const DropdownWithIcon = forwardRef((props, forwardedRef) => ...

// then...
<Dropdown.Button appearance={props.appearance}
```

And 

```
const context = useDropdown({ ... })

// then...
<Dropdown.Menu ref={context.menuRef} />
```

In other words, I think the style of destructuring every property out of giant objects leads to this problem.